src/journal/src/Journal/Internal/ByteBufferPtr.hs

```diff
@@ -68,6 +68,16 @@ newByteBuffer fptr capa lim pos mSli
   <*> newIORef (-1)
   <*> maybe (newIORef 0) return mSli
 
+fromBS :: BS.ByteString -> IO ByteBuffer
+fromBS bs
+  = ByteBuffer fptr (Capacity len)
+  <$> newIORef (Limit len)
+  <*> newIORef 0
+  <*> newIORef (-1)
+  <*> newIORef (Slice offset)
+  where
+    (fptr, offset, len) = BS.toForeignPtr bs
+
```

Comment:
This seems quite unsafe?
```haskell
*Journal> x = "hello" :: ByteString
*Journal> x
"hello"
*Journal> y <- fromBS x
*Journal> clean y
*Journal> x
"\NUL\NUL\NUL\NUL\NUL"
```

Reply:
Prefixed the name of the function with `unsafe` and added a comment that it was a hack.